borg/archiver.py

```diff
@@ -405,86 +406,148 @@ def item_is_hardlink_master(item):
-                    print('\t', args.location.archive, 'user=%s, group=%s' % (user1, group1))
-                    print('\t', args.archive2, 'user=%s, group=%s' % (user2, group2))
-
-                if not stat.S_ISREG(item1[b'mode']):
-                    return
-            if b'chunks' not in item1 or b'chunks' not in item2:
-                # At least one of the items is a link
-                if item1.get(b'source') != item2.get(b'source'):
-                    print(remove_surrogates(path), 'different link')
-                    print('\t', args.location.archive, item1.get(b'source', '<regular file>'))
-                    print('\t', args.archive2, item2.get(b'source', '<regular file>'))
-                return
-            if deleted or not can_compare_chunk_ids or item1[b'chunks'] != item2[b'chunks']:
-                # Contents are different
-                chunk_ids1 = [c[0] for c in item1[b'chunks']]
-                chunk_ids2 = [c[0] for c in item2[b'chunks']]
-                chunk_id_set1 = set(chunk_ids1)
-                chunk_id_set2 = set(chunk_ids2)
-                total1 = None if item1.get(b'deleted') else sum(c[1] for c in item1[b'chunks'])
-                total2 = None if item2.get(b'deleted') else sum(c[1] for c in item2[b'chunks'])
-                if (not can_compare_chunk_ids and total1 == total2 and not deleted and
-                        fetch_and_compare_chunks(chunk_ids1, chunk_ids2, archive1, archive2)):
-                    return
-                added = sum(c[1] for c in (chunk_id_set2 - chunk_id_set1))
-                removed = sum(c[1] for c in (chunk_id_set1 - chunk_id_set2))
-                print(remove_surrogates(path), 'different contents')
-                print('\t +%s, -%s, %s, %s' % (format_bytes(added), format_bytes(removed),
-                                               format_bytes(total1), format_bytes(total2)))
+                changes.append(compare_owner(item1, item2))
+                changes.append(compare_mode(item1, item2))
+
+            changes = [x for x in changes if x]
+            if changes:
+                print("{:<19} {}"
+                      .format(' '.join(changes), remove_surrogates(path)))
 
         def compare_archives(archive1, archive2, matcher):
-            orphans_archive1 = {}
-            orphans_archive2 = {}
+            orphans_archive1 = collections.OrderedDict()
+            orphans_archive2 = collections.OrderedDict()
+            hardlink_masters = {}
             for item1, item2 in zip_longest(
                     archive1.iter_items(lambda item: matcher.match(item[b'path'])),
                     archive2.iter_items(lambda item: matcher.match(item[b'path'])),
             ):
                 if item1 and item2 and item1[b'path'] == item2[b'path']:
-                    compare_items(item1[b'path'], item1, item2)
+                    compare_items(item1[b'path'], item1, item2,
+                                  hardlink_masters)
```

Comment:
hmm, are you doing these linebreaks to obey pep8 80chars limit?
we do not follow that in borg project, having lines with up to 120 chars is fine.

can you reformat the places where you introduced "80chars linebreaks"?